src/commands/pczt/create_manual.rs

```diff
@@ -0,0 +1,416 @@
+#![allow(deprecated)]
+use std::str::FromStr;
+
+use anyhow::anyhow;
+use clap::Args;
+use pczt::roles::{creator::Creator, io_finalizer::IoFinalizer, updater::Updater};
+use rand::rngs::OsRng;
+use serde::Deserialize;
+use tokio::io::{stdout, AsyncWriteExt};
+use transparent::{
+    address::TransparentAddress,
+    bundle::{OutPoint, TxOut},
+};
+
+use zcash_address::ZcashAddress;
+use zcash_client_backend::proto::service::{ChainSpec, TxFilter};
+use zcash_keys::address::{Address, Receiver};
+use zcash_primitives::transaction::{
+    builder::{Builder, PcztResult},
+    fees::zip317,
+    Transaction,
+};
+use zcash_protocol::{
+    consensus::{self, Parameters},
+    memo::{Memo, MemoBytes},
+    value::Zatoshis,
+};
+use zcash_script::script::{self, Parsable};
+
+use crate::{
+    config::WalletConfig,
+    data::Network,
+    error,
+    remote::{tor_client, Servers},
+};
+
+// Options accepted for the `pczt create-manual` command
+#[derive(Debug, Args)]
+pub(crate) struct Command {
+    /// The transparent coins to spend in this transaction
+    #[arg(long)]
+    coins: String,
+
+    /// The recipient's Unified, Sapling or transparent address
+    #[arg(long)]
+    address: String,
+
+    /// The amount in zatoshis
+    #[arg(long)]
+    value: Option<u64>,
+
+    /// A memo to send to the recipient
+    #[arg(long)]
+    memo: Option<String>,
+
+    /// The network the coins are from: \"test\" or \"main\".
+    ///
+    /// If unset, uses the network of the provided wallet.
+    #[arg(short, long)]
+    #[arg(value_parser = Network::parse)]
+    network: Option<Network>,
+
+    /// The server to use for network information (default is \"ecc\")
+    #[arg(short, long)]
+    #[arg(default_value = "ecc", value_parser = Servers::parse)]
+    server: Servers,
+
+    /// Disable connections via TOR
+    #[arg(long)]
+    disable_tor: bool,
+}
+
+fn parse_coins(s: &str) -> anyhow::Result<Vec<Coin>> {
+    Ok(serde_json::from_str(s)?)
+}
+
+#[derive(Clone, Debug, Deserialize)]
+struct Coin {
+    hash: String,
```

Comment:
Is this the txid of the outpoint? I know that "hash" and "n" are the names from the bitcoin sources, but they're lousy names.

Reply:
It's the txid of the transaction containing the coin as a transparent output.